File: src/Chapter-2-Rust/dweight2.rs
use std::process::Command;
use std::io;
use std::str::FromStr;
//Computes the dimensional weight of a 12 x 10 x 8 box

fn main()
{
	//Initialise variables for string
	//If I used just one string variable for each user input
	//it crashed. This feels like a poor solution
	let mut string = String::new();
	
	//Get user input
	println!("Enter height of the box: ");
	io::stdin().read_line(&mut string).ok();
	let height = i32::from_str(string.trim()).unwrap();
	string.clear();

	println!("Enter length of box: ");
	io::stdin().read_line(&mut string).ok();
	let length = i32::from_str(string.trim()).unwrap();
	string.clear();
	
	println!("Enter width of box: ");
	io::stdin().read_line(&mut string).ok();	
	let width = i32::from_str(string.trim()).unwrap();

	//Calculate volume
	let volume = height * length * width;
	//Calculate weight
	let weight = (volume + 165) / 166;
	//Print details
	println!("Volumne (cubic inches): {}", volume);
	println!("Dimensional weight (pounds: {}", weight);
	
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}

File: src/Chapter-2-C99/dweight2.c
/* Computes the dimensional weight of a
   box from input provided by the user */

#include <stdio.h>

int main(void)
{
    //Initialise variables
    int height, length, width, volume, weight;
    //User input - why the '&' symbol?
    printf("Enter height of box: ");
    scanf("%d", &height);
    printf("Enter length of box: ");
    scanf("%d", &length);
    printf("Enter width of box: ");
    scanf("%d", &width);
    //Calculate volume
    volume = height * length * width;
    //Calculate weight
    weight = (volume + 165) / 166;
    //Print details
    printf("Volume (cubic inches): %d\n", volume);
    printf("Dimensional weight (pounds): %d\n", weight);
    return 0;

}
